pkg/operator/ceph/controller/cluster_info.go

```diff
@@ -124,9 +130,32 @@ func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.
 			MonitorSecret: string(secrets.Data[MonSecretNameKey]),
 			Context:       context,
 		}
+
 		if cephUsername, ok := secrets.Data[CephUsernameKey]; ok {
 			clusterInfo.CephCred.Username = string(cephUsername)
 			clusterInfo.CephCred.Secret = string(secrets.Data[CephUserSecretKey])
+			// for upgrade cluster get rookoperator keyrings
+			if string(cephUsername) == "client.admin" {
+
+				_, err := genAdminKeyring(clusterdContext, namespace)
```

Comment:
No need for `genAdminKeyring()`, the call to `AuthGetOrCreateKey()` would generate the new keyring.